play-json-tools/src/test/scala/com/evolutiongaming/playjsontools/DiscriminatedEitherFormatSpec.scala

```diff
@@ -0,0 +1,36 @@
+package com.evolutiongaming.playjsontools
+
+import com.evolutiongaming.playjsontools.PlayJsonHelper._
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+import play.api.libs.json._
+
+class DiscriminatedEitherFormatSpec extends AnyFunSuite with Matchers {
+  implicit val eitherFormat = DiscriminatedEitherFormat.eitherFormat("L", "R"): OFormat[Either[String, Int]]
+  val left: Either[String, Int] = Left("foo")
+  val right: Either[String, Int] = Right(1)
+  val fromEitherJson: JsValue => JsResult[Either[String, Int]] = Json.fromJson[Either[String, Int]](_)
+
+  test("eitherFormat left") {
+    val value = left
+    val json = Json.toJson(value)
+    json shouldEqual Json.obj("L" -> "foo")
+    fromEitherJson(json) shouldEqual JsSuccess(value)
+  }
+
+  test("eitherFormat right") {
+    val value = right
+    val json = Json.toJson(value)
+    json shouldEqual Json.obj("R" -> 1)
+    fromEitherJson(json) shouldEqual JsSuccess(value)
+  }
+
+  test("eitherFormat should fail on non-objects") {
+    fromEitherJson(JsNull) shouldEqual JsError("can't parse either for non-object for null")
+  }
+
+  test("eitherFormat should fail on extra keys") {
```

Comment:
what's about case of ?

```
{"R": 1, "metadata": "metadata"}
```

Reply:
Would you expect it to work? Why?

Reply:
because usually when we deal with json we are parsing only needed parts and ignoring non-conflicting data :)

Reply:
Alright, that does seem to be the case with other decoders. Changed the code to fit the description.